runtime/vm/extendedMessageNPE.cpp

```diff
@@ -159,19 +372,23 @@ getCompleteNPEMessage(J9VMThread *vmThread, U_8 *bcCurrentPtr, J9ROMClass *romCl
 			Trc_VM_GetCompleteNPEMessage_Invoke_Index(vmThread, index);
 
 			J9ROMMethodRef *romMethodRef = (J9ROMMethodRef *)&constantPool[index];
-			J9ROMNameAndSignature *nameAndSig = J9ROMMETHODREF_NAMEANDSIGNATURE(romMethodRef);
-			J9UTF8 *sig = J9ROMNAMEANDSIGNATURE_SIGNATURE(nameAndSig);
-			J9UTF8 *name = J9ROMNAMEANDSIGNATURE_NAME(nameAndSig);
-			const char *VM_NPE_INVOKEMETHOD = "Cannot invoke \"%2$.*1$s.%4$.*3$s%6$.*5$s\"";
-			J9UTF8 *definingUTF = J9ROMSTRINGREF_UTF8DATA((J9ROMStringRef *)&constantPool[romMethodRef->classRefCPIndex]);
-
-			msgLen = j9str_printf(PORTLIB, NULL, 0, VM_NPE_INVOKEMETHOD,
-				J9UTF8_LENGTH(definingUTF), J9UTF8_DATA(definingUTF),
-				J9UTF8_LENGTH(name), J9UTF8_DATA(name), J9UTF8_LENGTH(sig), J9UTF8_DATA(sig));
-			npeMsg = (char *)j9mem_allocate_memory(msgLen + 1, OMRMEM_CATEGORY_VM);
-			j9str_printf(PORTLIB, npeMsg, msgLen, VM_NPE_INVOKEMETHOD,
-				J9UTF8_LENGTH(definingUTF), J9UTF8_DATA(definingUTF),
-				J9UTF8_LENGTH(name), J9UTF8_DATA(name), J9UTF8_LENGTH(sig), J9UTF8_DATA(sig));
+			J9ROMNameAndSignature *methodNameAndSig = J9ROMMETHODREF_NAMEANDSIGNATURE(romMethodRef);
+			J9UTF8 *methodName = J9ROMNAMEANDSIGNATURE_NAME(methodNameAndSig);
+			const char *VM_NPE_INVOKEMETHOD = "Cannot invoke \"%s.%.*s%s\"";
+
+			J9UTF8 *definingClassFullQualifiedName = J9ROMSTRINGREF_UTF8DATA((J9ROMStringRef *)&constantPool[romMethodRef->classRefCPIndex]);
+			char *fullyQualifiedDefiningClassName = convertToJavaFullyQualifiedName(vmThread, definingClassFullQualifiedName);
+			char *methodSigParameters = convertMethodSignature(vmThread, J9ROMNAMEANDSIGNATURE_SIGNATURE(methodNameAndSig));
+
+			if ((NULL != definingClassFullQualifiedName) && (NULL != methodSigParameters)) {
+				msgLen = j9str_printf(PORTLIB, NULL, 0, VM_NPE_INVOKEMETHOD,
+					fullyQualifiedDefiningClassName, J9UTF8_LENGTH(methodName), J9UTF8_DATA(methodName), methodSigParameters);
+				npeMsg = (char *)j9mem_allocate_memory(msgLen + 1, OMRMEM_CATEGORY_VM);
```

Comment:
Missing NULL check.

Reply:
The `NULL` check was skipped because `j9str_printf` accepts `NULL` and a `NULL` message will be returned instead of attempting to create another different error message (other than current NPE).
There is a comment for earlier `j9mem_allocate_memory` call https://github.com/eclipse/openj9/blob/01c213f220a1e86c2977a23becf7cf7206bbe7a3/runtime/vm/extendedMessageNPE.cpp#L85